src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -24753,4 +24752,419 @@ public static void M<U>(U u) { }
+        Assert.Equal("void E.<>E__0.M2()", model.GetSymbolInfo(invocation).Symbol.ToTestDisplayString());
+
+        invocation = GetSyntax<InvocationExpressionSyntax>(tree, "M2(new object())");
+        Assert.Equal("void E.M2(this System.Object o)", model.GetSymbolInfo(invocation).Symbol.ToTestDisplayString());
+    }
+
+    [Fact]
+    public void This_03()
+    {
+        var src = """
+static class E
+{
+    extension(object o)
+    {
+        public static void M()
+        {
+            M2();
+        }
+        public void M2() { }
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics(
+            // (7,13): error CS7036: There is no argument given that corresponds to the required parameter 'o' of 'E.M2(object)'
+            //             M2();
+            Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, "M2").WithArguments("o", "E.M2(object)").WithLocation(7, 13));
+    }
+
+    [Fact]
+    public void This_04()
+    {
+        var src = """
+static class E
+{
+    extension(object o)
+    {
+        public void M()
+        {
+            M2();
+        }
+        public static void M2() { }
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics();
+
+        var tree = comp.SyntaxTrees.First();
+        var model = comp.GetSemanticModel(tree);
+        var invocation = GetSyntax<InvocationExpressionSyntax>(tree, "M2()");
+        Assert.Equal("void E.M2()", model.GetSymbolInfo(invocation).Symbol.ToTestDisplayString());
+    }
+
+    [Fact]
+    public void ReceiverParameter_Access_01()
+    {
+        var src = """
+static class E
+{
+    extension(object o)
+    {
+        public void M()
+        {
+            o.M2();
+        }
+        public void M2() { }
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics();
+    }
+
+    [Fact]
+    public void ReceiverParameter_Access_02()
+    {
+        var src = """
+static class E
+{
+    extension(object o)
+    {
+        public void M()
+        {
+            o.M2();
+        }
+        public static void M2() { }
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics(
+            // (7,13): error CS0176: Member 'E.extension(object).M2()' cannot be accessed with an instance reference; qualify it with a type name instead
+            //             o.M2();
+            Diagnostic(ErrorCode.ERR_ObjectProhibited, "o.M2").WithArguments("E.extension(object).M2()").WithLocation(7, 13));
+    }
+
+    [Fact]
+    public void ReceiverParameter_Access_03()
+    {
+        var src = """
+static class E
+{
+    extension(object o)
+    {
+        public static void M()
+        {
+            o.M2();
+        }
+        public void M2() { }
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics(
+            // (7,13): error CS0103: The name 'o' does not exist in the current context
```

Comment:
>// (7,13): error CS0103: The name 'o' does not exist in the current context [](http://example.com/codeflow?start=12&length=75)

This test is going to cause a conflict with #77704

Reply:
Yup. I'll merge/resolve after your PR goes in